crates/swift-bridge-ir/src/codegen/generate_rust_tokens/shared_enum.rs

```diff
@@ -140,22 +140,27 @@ impl SwiftBridgeModule {
         // User derives
         let mut derive_impl_ffi_bridges = vec![];
 
-        // We currently only allow derive(Debug) on non data carrying enums in order
-        // to prevent a potential memory safety issue.
-        // https://github.com/chinedufn/swift-bridge/pull/194#discussion_r1134386788
-        if shared_enum.derive.debug && !shared_enum.has_one_or_more_variants_with_data() {
+        if shared_enum.derive.debug {
+            // We don't want to confuse the developer if one of our variants has data and Debug isn't derived,
+            // so we still want to derive(Debug) on the Rust side.
```

Comment:
You can remove this first part of the comment.

We don't need to explain why we always emit the `Debug` impl on the Rust side. This is desirable behavior.

Can keep the TODO. It's worth considering.